src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/FlowAnalysis/SymbolUsageAnalysis/SymbolUsageAnalysis.BasicBlockAnalysisData.cs

```diff
@@ -112,20 +112,16 @@ public bool Equals(BasicBlockAnalysisData other)
                     return false;
                 }
 
-                using var _ = PooledHashSet<ISymbol>.GetInstance(out var uniqueSymbols);
-
-                // Check if both _reachingWrites maps have same set of keys.
-#if NET
-                uniqueSymbols.EnsureCapacity(Math.Max(_reachingWrites.Keys.Count, other._reachingWrites.Keys.Count));
-#endif
-
-                uniqueSymbols.AddRange(_reachingWrites.Keys);
-                uniqueSymbols.AddRange(other._reachingWrites.Keys);
-                if (uniqueSymbols.Count != _reachingWrites.Count)
-                    return false;
+                // Check if both _reachingWrites maps have same set of keys.  This is a quick out based on O(keys),
+                // instead of doing the full O(k*v) check below.
+                foreach (var key in _reachingWrites.Keys)
+                {
+                    if (!other._reachingWrites.ContainsKey(key))
+                        return false;
+                }
```

Comment:
still written as a handwritten look at there aren't any helpers i can find that can do the same without an allocation.